mesh/meshdb.go

```diff
@@ -443,25 +456,31 @@ func (m *DB) WriteTransaction(l types.LayerID, t *types.Transaction) error {
 	return nil
 }
 
+//Reason we're not using the existing reward type because the layer is implicit in the key
 type dbReward struct {
 	TotalReward         uint64
 	LayerRewardEstimate uint64
+	SmesherID           types.NodeID
+	Coinbase            types.Address
 	// TotalReward - LayerRewardEstimate = FeesEstimate
 }
 
-func (m *DB) writeTransactionRewards(l types.LayerID, accounts []types.Address, totalReward, layerReward *big.Int) error {
-	actBlockCnt := make(map[types.Address]uint64)
-	for _, account := range accounts {
-		actBlockCnt[account]++
-	}
+func (m *DB) writeTransactionRewards(l types.LayerID, accountBlockCount map[types.Address]uint64, totalReward, layerReward *big.Int, smeshers map[types.Address]types.NodeID) error {
+	// actBlockCnt := make(map[types.Address]uint64)
+	// for _, account := range accounts {
+	// 	actBlockCnt[account]++
+	// }
 
 	batch := m.transactions.NewBatch()
-	for account, cnt := range actBlockCnt {
-		reward := dbReward{TotalReward: cnt * totalReward.Uint64(), LayerRewardEstimate: cnt * layerReward.Uint64()}
+	for account, cnt := range accountBlockCount {
+		smesherID := smeshers[account]
+		reward := dbReward{TotalReward: cnt * totalReward.Uint64(), LayerRewardEstimate: cnt * layerReward.Uint64(), SmesherID: smesherID, Coinbase: account}
 		if b, err := types.InterfaceToBytes(&reward); err != nil {
 			return fmt.Errorf("could not marshal reward for %v: %v", account.Short(), err)
 		} else if err := batch.Put(getRewardKey(l, account), b); err != nil {
 			return fmt.Errorf("could not write reward to %v to database: %v", account.Short(), err)
+		} else if err := batch.Put(getSmesherRewardKey(l, smesherID), getRewardKey(l, account)); err != nil {
```

Comment:
See note above about `getRewardKeyPrefix` conflicting with `getSmesherRewardKey`. I don't think you need both here.